src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/FaultDetectionMetricsProcessor.java

```diff
@@ -0,0 +1,142 @@
+                emitStartMetric(event, keyItems);
+            } else if (keyItems[3].equals(PerformanceAnalyzerMetrics.FINISH_FILE_NAME)) {
+                emitFinishMetric(event, keyItems);
+            }
+        }
+    }
+
+    @Override
+    public boolean shouldProcessEvent(Event event) {
+        return event.key.contains(PerformanceAnalyzerMetrics.sFaultDetection);
+    }
+
+    @Override
+    public void commitBatchIfRequired() {
+        if (handle.size() > BATCH_LIMIT) {
+            handle.execute();
+            handle = faultDetectionMetricsSnapshot.startBatchPut();
+        }
+    }
+
+    // A keyItem is of the form : [fault_detection, follower_check, 76532, start]
+    //
+    // Example value part of the entry is:
+    // current_time:1566413979979
+    // StartTime:1566413987986
+    // SourceNodeID:g52i9a93a762cd59dda8d3379b09a752a
+    // TargetNodeID:b2a5a93a762cd59dda8d3379b09a752a
+    // $
+    private void emitStartMetric(Event entry, String[] keyItems) {
+        Map<String, String> keyValueMap = ReaderMetricsProcessor.extractEntryData(entry.value);
+
+        String sourceNodeId = keyValueMap.get(AllMetrics.FaultDetectionDimension.SOURCE_NODE_ID.toString());
+        String targetNodeId = keyValueMap.get(AllMetrics.FaultDetectionDimension.TARGET_NODE_ID.toString());
+        String startTimeVal = keyValueMap.get(AllMetrics.CommonMetric.START_TIME.toString());
+        String fault_detection_type = keyItems[1];
+        try {
+            long st = Long.parseLong(startTimeVal);
+            // A keyItem is of the form : [fault_detection, follower_check, 76543, start]
+            String rid = keyItems[2];
+            handle.bind(rid, sourceNodeId, targetNodeId, fault_detection_type, st, null, 0);
+        } catch (NumberFormatException e) {
+            LOG.error("Unable to parse string. StartTime:{}", startTimeVal);
+            StatsCollector.instance().logException(StatExceptionCode.READER_PARSER_ERROR);
+            throw e;
+        }
+    }
+
+    // A keyItem is of the form : [fault_detection, follower_check, 76532, finish]
+    //
+    // Example value part of the entry is:
+    // current_time:1566413979979
+    // FinishTime:1566413987986
+    // SourceNodeID:g52i9a93a762cd59dda8d3379b09a752a
+    // TargetNodeID:b2a5a93a762cd59dda8d3379b09a752a
+    // Error:0
+    // $
+    private void emitFinishMetric(Event entry, String[] keyItems) {
+        Map<String, String> keyValueMap = ReaderMetricsProcessor.extractEntryData(entry.value);
+
+        String sourceNodeId = keyValueMap.get(AllMetrics.FaultDetectionDimension.SOURCE_NODE_ID.toString());
+        String targetNodeId = keyValueMap.get(AllMetrics.FaultDetectionDimension.TARGET_NODE_ID.toString());
+        String finishTimeVal = keyValueMap.get(AllMetrics.CommonMetric.FINISH_TIME.toString());
+        String errorString = keyValueMap.get(PerformanceAnalyzerMetrics.ERROR);
+        String fault_detection_type = keyItems[1];
```

Comment:
similar, move this to within try block after line 133

Reply:
done